lib/maths/CBoostedTreeLoss.cc

```diff
@@ -366,6 +366,144 @@ CArgMinMultinomialLogisticLossImpl::objectiveGradient() const {
         return TDoubleVector{2.0 * lambda * weight + lossGradient};
     };
 }
+
+CArgMinMsleImpl::CArgMinMsleImpl(double lambda)
+    : CArgMinLossImpl{lambda}, m_Buckets(128) {
```

Comment:
If I reduce the bucket size, I would need to relax the test containts since `CBoostedTreeLossTest/testMsleArgminObjective` starts to fail then.

Reply:
I think this is ok. For example, I'm a bit worried about the memory this will use for a large tree. We don't actually monitor this memory at present (something we should perhaps do in a follow on PR).

Reply:
I changed the bucket size to 32 and relaxed the unit test limits.